### PR TITLE
Honor blank "additional content" in e-mails (vs using default text)

### LINKED ISSUE #41462: Blank "Additional content" in emails uses default text, rather than removing
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Under each email type in WooCommerce -> Settings -> Emails, there exists an "Additional content" field to add text to emails.
By default, these include phrases like "Congratulations on the sale" and "Thanks for reading".
These phrases are not welcomed in most scenarios and the removal of them used to be as simple as deleting the text from the field.

More recently, having a blank "Additional content" field, a hard-coded default option is used regardless.

This can be seen in the code at https://github.com/woocommerce/woocommerce/blob/503dad4b57029cb2d1b459ff6c4c1bc8b485763a/plugins/woocommerce/includes/emails/class-wc-email.php#L419C26-L419C63

Specifically:
`$this->get_option( 'additional_content', $this->get_default_additional_content())`

### Expected behavior

When the "Additional content" field is blank, do not show any additional content

### Actual behavior

When the "Additional content" field is blank, a default text is used

### Steps to reproduce

1. Remove content to "Additional content" field within an email configuration (e.g. WC->Settings->Emails->New Order)
2. Send an email
3. See default "Additional content" text shown

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by jorgeatorres**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Currently, e-mails that have "additional content" (most do) will use the default value set in the e-mail class even when that additional content setting has been modified by the merchant to be empty. This means there's currently no way to have no additional content in those e-mails.

It seems to me we should honor what the merchant has configured and the default value should only be used when the merchant hasn't customized the e-mail at all.

Closes #41462.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Have a way to "catch" e-mails sent by your WP instance. I recommend MailHog or the WP Mail Logging plugin.
2. Either manually create an order in WC > Orders or make a purchase through the frontend.
3. Go to WC > Settings > E-mails and choose **Order details**.
4. Modify **Additional content** to anything of your liking and save.
5. Go to WC > Orders and choose the order created in step 2.
6. Choose **Send order details to customer** and click the action button **`>`**.
7. Using your mail catcher, confirm that the e-mail is sent and contains the additional content you've specified.
8. Repeat 3-4 but this time remove all content from **Additional content**.
9. Repeat steps 5-7 and confirm no additional content is sent.
   **Optional:** Contrast this with `trunk`, where despite this customization, the e-mail will contain the default text of "Thank you for using [site URL]!".
11. Run `wp option patch delete woocommerce_customer_invoice_settings additional_content` to simulate the e-mail not having been customized by the merchant.
12. Repeat steps 5-7 and confirm that the e-mail contains the default message of "Thank you for using [site URL]!".

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by barryhughes** [comment]
If handy when testing this/similar things, new order emails can be triggered repeatedly with this WP CLI one-liner (where `1234` is a valid order ID):

```sh
wp eval "add_filter( 'woocommerce_new_order_email_allows_resend', '__return_true' ); ( new WC_Email_New_Order )->trigger( 1234 );"
```

Should be easy to adapt to other email types.

File: plugins/woocommerce/includes/emails/class-wc-email.php [hunk 1]
```diff
@@ -407,6 +407,11 @@ public function get_default_additional_content() {
 	 * @return string
 	 */
 	public function get_additional_content() {
+		$additional_content = $this->get_option( 'additional_content' );
+		if ( ! is_string( $additional_content ) ) {
```
Comment (barryhughes) [inline plugins/woocommerce/includes/emails/class-wc-email.php]: One thing I'm curious about is when this condition that tests if `$additional_content` is not a string might be satisfied (or if it is needed, or if it makes sense).

Suppose the merchant has never touched the email settings, and they then manually send the order details to the customer: the call on the previous line (to `$this->get_option( 'additional_content' )`) is enough to fetch the default additional content without this block.

**Review by barryhughes — COMMENTED** [review]
Nice, works as per testing instructions and I think this makes sense overall.

One quibble (though almost too silly to mention) is it doesn't seem easy for the merchant to get back to the defaults (suppose they delete the additional content field in the email settings, then decide they want the default back), though of course they can manually reconstruct it.

I also left a question to clarify how a piece of code works—curious if we can possibly remove a few extra lines.

File: plugins/woocommerce/includes/emails/class-wc-email.php (hunk 1 above)
Comment (jorgeatorres) [reply plugins/woocommerce/includes/emails/class-wc-email.php]: You're right. I was doing some testing on the defaults `$this->get_option()` uses (not just for the "additional content" setting) but in general and probably got confused. That is, for all other e-mail settings (subject, heading, etc.), the default is an empty string (`''`) but for "additional content" the default is... well... the default additional content. That's a bit confusing as we do want to use the default for the other settings when the field is empty but not for the additional content.

Anyways, I've simplified things in a27815bbee01b96701afbe28f7ced83b3e4a334. Hope that's what you meant.

**Comment by jorgeatorres** [comment]
Hey @barryhughes!

Thanks for the review. I've made some changes and this is ready for another look. Please let me know what you think.

> One quibble (though almost too silly to mention) is it doesn't seem easy for the merchant to get back to the defaults (suppose they delete the additional content field in the email settings, then decide they want the default back), though of course they can manually reconstruct it.

Yeah, you're right and ideally we would have a way to "restore" the default value for all fields, not just the additional content field. Maybe we can work on that, though it feels like material for a different PR. wdyt?

**Comment by barryhughes** [comment]
> Maybe we can work on that, though it feels like material for a different PR. wdyt

Yep, agree. 

I included that note for completeness, but it doesn't feel like this is something people are going to trip over in the real world, so let's wait until/if someone requests it, and discuss further then.

File: plugins/woocommerce/includes/emails/class-wc-email.php (hunk 1 above)
Comment (barryhughes) [reply plugins/woocommerce/includes/emails/class-wc-email.php]: Yes, that's exactly it (in fact, it's even cleaner/more concise than I was thinking). Thank you!

**Review by barryhughes — APPROVED** [review]

**Pull request merged**